frontend/src/components/JqueryQueryBuilder/JqueryQueryBuilder.tsx

```diff
@@ -32,9 +36,38 @@ const JqueryQueryBuilder: FunctionComponent<JqueryQueryBuilderProps> = ({
     jq.init();
 
     if (Object.keys(rules).length > 0) {
+      if (
+        rules &&
+        Object.prototype.toString.call(
+          ((rules as JqueryQueryBuilderHaving)['rules'] as JqueryQueryBuilderHavingComparator[])[0]
+            .value,
+        ) === '[object Object]'
+      ) {
+        const element = (
+          (rules as JqueryQueryBuilderHaving)['rules'] as JqueryQueryBuilderHavingComparator[]
+        )[0];
+        if ('plain' in (element.value as { plain: string | number })) {
+          element.value = (element.value as { plain: string | number })['plain'];
+        } else {
+          const previousValue: { column: string; aggregate: string } = element.value as {
+            column: string;
+            aggregate: string;
+          };
+          delete element.value;
+          element.values = sortAggregateOptions(
+            aggregateOptions,
+            (previousValue as { column: string; aggregate: string }).aggregate,
+          ).map((option) => {
+            return {
+              value: `${previousValue?.column},${option}`,
+              label: `${option}(${previousValue?.column})`,
+            };
+          });
+        }
+      }
+      console.log(rules);
```

Comment:
Please do this outside of this component... it's what props are for cos this is crowding what was initially a simple component.